docs/plugins/plugin-writer/first-plugin.rst

```diff
@@ -1,2 +1,135 @@
+  * instance of :class:`~pulpcore.plugin.models.ContentArtifact` to be able to create relations with the artifact models.
+  * instance of :class:`~pulpcore.plugin.models.RemoteArtifact` to store information about artifact from remote source and to make a relation with  :class:`~pulpcore.plugin.models.ContentArtifact` created before.
+
+ * If a remote content should be downloaded right away (aka ``immediate`` download policy), use one of the suggested download solutions (:ref:`concurrent.futures <futures-docs>` or :ref:`asyncio <asyncio-docs>` approach) to start downloading. If content should be downloaded later (aka ``on_demand`` or ``background`` download policy), feel free to skip this step.
+ * Save all artifact and content data in one transaction:
+
+  * in case of downloaded content, create instance of :class:`~pulpcore.plugin.models.Artifact` which refers to a downloaded file on a filesystem and contains calculated checksums for it.
+  * in case of downloaded content, update :class:`~pulpcore.plugin.models.ContentArtifact` with reference to the created :class:`~pulpcore.plugin.models.Artifact`.
+  * create and save instance of :class:`~pulpcore.plugin.models.RepositoryContent` to associate content to a repository.
+  * save all created artifacts and content: ``ExampleContent``, :class:`~pulpcore.plugin.models.ContentArtifact`, :class:`~pulpcore.plugin.models.RemoteArtifact`.
+
+ * Use :class:`~pulpcore.plugin.models.ProgressBar` to report progress of some steps if needed.
+
+
+There are several important aspects relevant to importer implementation which were briefly mentioned in :ref:`understanding-models` section:
+
+ * due to deduplication of :class:`~pulpcore.plugin.models.Content` and :class:`~pulpcore.plugin.models.Artifact` data, they may already exist and the importer needs to fetch and use them when they do.
+ * :class:`~pulpcore.plugin.models.ContentArtifact` associates :class:`~pulpcore.plugin.models.Content` and :class:`~pulpcore.plugin.models.Artifact`. If :class:`~pulpcore.plugin.models.Artifact` is not downloaded yet, :class:`~pulpcore.plugin.models.ContentArtifact` contains ``NULL`` value for :attr:`~pulpcore.plugin.models.ContentArtifact.artifact`. It should be updated whenever corresponding :class:`~pulpcore.plugin.models.Artifact` is downloaded.
+
+Check our Example plugin implementation of importers: `one uses asyncio <https://github.com/pulp/pulp_example/blob/master/pulp_example/app/models.py#L529-L833>`_ as a solution for downloading, `the other one follows concurrent.futures approach <https://github.com/pulp/pulp_example/blob/master/pulp_example/app/models.py#L184-L526>`_.
+
+The importer implementation suggestion above allows plugin writer to have an understanding and control at a low level.
+Pulp team also offers plugin writers to try out a higher level, more simplified, API which introduces the concept of :class:`~pulpcore.plugin.changeset.ChangeSet`.
```

Comment:
What about s/Pulp team also offers plugin writers to try out/The plugin API has?